packages/bpk-component-button/src/BpkButtonV2/common-types.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactNode, SyntheticEvent } from 'react';
+
+export const BUTTON_TYPES = {
+  primary: 'primary',
+  primaryOnDark: 'primary-on-dark',
+  primaryOnLight: 'primary-on-light',
+  secondary: 'secondary',
+  secondaryOnDark: 'secondary-on-dark',
+  destructive: 'destructive',
+  featured: 'featured',
+  link: 'link',
+  linkOnDark: 'link-on-dark',
+} as const;
+
+export const SIZE_TYPES = {
+  small: 'small',
+  large: 'large',
+} as const;
+
+export type ButtonType = typeof BUTTON_TYPES[keyof typeof BUTTON_TYPES];
```

Comment:
I think this would actually have to be the values, not the keys of BUTTON_TYPES as the class names are using hyphens

Reply:
Good point this was just me copying from the Badges

Reply:
I did a check of this in both Banana and online, this is correct for getting the values to be expected for the type :)